### PR TITLE
BCDA-4129 - Replace lib/pq driver with pgx3

### PR DESCRIPTION
**Pull request by mtrang1263**
### Fixes [BCDA-4129](https://jira.cms.gov/browse/BCDA-4129)

Per [lib/pq](https://github.com/lib/pq#status), they recommend switching over to pgx. With pgx, we get more modern conveniences, especially if/when we move to v4.

### Change Details
1. Update references to lib/pq and replace them with pgx.
2. When needed, use an unused import (`_ "github.com/jackc/pgx/stdlib"`) to ensure that we have pgx properly loaded.
3. Update our CCLF ingest to use pgx.CopyFrom. This used pgx.pgTypes directly based on this [issue](https://github.com/jackc/pgx/issues/763).

### Security Implications

- [x] new software dependencies
    Removed lib/pq dependency.
- [ ] security controls or supporting software altered
- [ ] new data stored or transmitted
- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
1. CI Passes.
2. Deploy to dev and verify that CCLF ingestion performance was not impacted with the change. Screenshot below shows ingestion times on dev. No noticeable increase.
Screen Shot 2021-01-27 at 9 43 36 AM
3. Smoke test passing verifying that postgres driver swap works on the remote environments.

### CODE REVIEW

File: bcda/cclf/cclf.go [hunk 1]
```diff
@@ -244,68 +238,22 @@ func importCCLF8(ctx context.Context, fileMetadata *cclfFileMetadata) (err error
 	sc := bufio.NewScanner(rc)
 
 	// Open transaction to encompass entire CCLF file ingest.
-	txn, err := db.Begin()
```
[the remark this answers is not in the document]
Comment (CosmicSyntax) [reply bcda/cclf/cclf.go:247]: Should we utilize the "VACUUM" if we do hit an error in the future?  If this is already done, disregard.

[the remark this answers is not in the document]
Comment (CosmicSyntax) [reply bcda/cclf/cclf.go:247]: nvm

File: bcda/cclf/importer.go [hunk 2]
```diff
@@ -1,83 +1,113 @@
...
-	if cclfImporter.pendingQueries >= cclfImporter.maxPendingQueries {
-		if err := cclfImporter.flush(ctx); err != nil {
-			return errors.Wrap(err, "failed to flush statement")
-		}
-		if err := cclfImporter.refreshStatement(ctx, tx); err != nil {
-			return errors.Wrap(err, "failed to refresh statement")
+		mbi := importer.getMBI()
+		// We've already processed this MBI before
+		if _, found := importer.processedMBIs[mbi]; found {
+			continue
```
Comment (CosmicSyntax) [inline bcda/cclf/importer.go:49]: If you think we should follow this pattern for existence of a value in map, we can change the rest of the code base to match.

File: bcda/cclf/cclf.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (mtrang1263) [reply bcda/cclf/cclf.go:247]: > Should we utilize the "VACUUM" if we do hit an error in the future? If this is already done, disregard.

We should and that's being taken care of with our postgres instances in AWS. We have AUTO VACUUM set.

File: bcda/cclf/importer.go (hunk 2 above)
Comment (mtrang1263) [reply bcda/cclf/importer.go:49]: I think it's a good pattern to use, but not so passionate that it needs to be enforced in all use cases. I find it clearer but understand that the map[interface{}]bool also makes sense for existence.

File: bcda/cclf/cclf.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (CosmicSyntax) [reply bcda/cclf/cclf.go:247]: Sweet! Thanks for the info.

**Review by CosmicSyntax — APPROVED** [review]

**Comment by CosmicSyntax** [comment]
LGTM

**Review by SMLuthi — APPROVED** [review]
LGTM!

**Pull request merged**